openerp/addons/base/ir/ir_attachment.py

```diff
@@ -263,11 +263,13 @@ def _compute_mimetype(self, values):
             :param values : dict of values to create or write an ir_attachment
             :return mime : string indicating the mimetype, or application/octet-stream by default
         """
-        mimetype = 'application/octet-stream'
+        mimetype = False
         if values.get('datas_fname'):
             mimetype = mimetypes.guess_type(values['datas_fname'])[0]
-        if values.get('datas'):
+        if not mimetype and values.get('datas'):
```

Comment:
The first condition `if values.get('datas_fname')` always True because without file name not attached file. so the always `return mimetype` but without file extension they attach `return None` that time we check `if not mimetype and values.get('datas')` attach file is not empty `retrun mimetype` if attach file is empty go to thrid condition `if not mimetype` `retrun mimetype`.